src/main/java/de/tum/in/www1/artemis/service/LectureUnitProcessingService.java

```diff
@@ -73,16 +93,67 @@ public List<LectureUnitDTO> splitUnits(LectureUnitInformationDTO lectureUnitInfo
                 attachment.setUploadDate(ZonedDateTime.now());
 
                 MultipartFile multipartFile = fileService.convertByteArrayToMultipart(lectureUnit.unitName(), ".pdf", outputStream.toByteArray());
+                AttachmentUnit savedAttachmentUnit = saveAttachmentUnit(attachmentUnit, attachment, multipartFile, lecture);
+                slideSplitterService.splitAttachmentUnitIntoSingleSlides(documentUnits.get(0), savedAttachmentUnit);
 
-                LectureUnitDTO lectureUnitsDTO = new LectureUnitDTO(attachmentUnit, attachment, multipartFile);
-                units.add(lectureUnitsDTO);
-                documentUnits.get(0).close(); // make sure to close the document
+                units.add(savedAttachmentUnit);
             }
+            lectureRepository.save(lecture);
             document.close();
             return units;
         }
     }
 
+    /**
+     * Save the attachment unit with.
+     *
+     * @param attachmentUnit The attachment unit to be saved
+     * @param attachment     The attachment to be saved
+     * @param multipartFile  The file to be saved
+     * @param lecture        The lecture that the attachment unit belongs to
+     * @return The saved attachment unit
+     */
+    private AttachmentUnit saveAttachmentUnit(AttachmentUnit attachmentUnit, Attachment attachment, MultipartFile multipartFile, Lecture lecture) {
+        attachmentUnit.setLecture(null);
+        AttachmentUnit savedAttachmentUnit = attachmentUnitRepository.saveAndFlush(attachmentUnit);
+        attachmentUnit.setLecture(lecture);
+        lecture.addLectureUnit(savedAttachmentUnit);
+
+        handleFile(multipartFile, attachment);
+
+        attachment.setAttachmentUnit(savedAttachmentUnit);
+        attachment.setVersion(1);
+
+        Attachment savedAttachment = attachmentRepository.saveAndFlush(attachment);
+        attachmentUnit.setAttachment(savedAttachment);
+        evictCache(multipartFile, savedAttachmentUnit);
+        return savedAttachmentUnit;
+    }
```

Comment:
This was part of `AttachmentUnitService ` but since now i have introduced the slide splitter i decided to move this logic into splitUnits so now as soon as attachment unit is split the attachment unit is saved and also most importantly the slide is split into single images.  The reason for this is that i had an issue when loading `MultipartFiles` as array into Documents the PDFBOX library was loading all the time the first file. I have researched this for more then 2 weeks and i couldn't find a solution (i also posted into slack channel). So basically i don't have to load the files into document inside a for loop (all the attachment units) and then split the slide into single images but when the attachment unit is saved i split the slide immediately.

Reply:
I see, but I don't see an explanation for why you decided to move it there. You save an attachment unit, so intuitively, it should be placed inside the corresponding service, shouldn't it? I don't think that the explanation justifies putting it in another service.

Reply:
I modified `createAttachmentUnit` in `AttachmentUnitService` and now i use this method for both cases create attachment and create multiple attachments.